src/main/kotlin/no/nav/familie/ef/sak/api/beregning/BeregningService.kt

```diff
@@ -1,17 +1,138 @@
 package no.nav.familie.ef.sak.api.beregning
 
+import no.nav.familie.ef.sak.api.feilHvis
+import no.nav.familie.ef.sak.util.Periode
+import no.nav.familie.ef.sak.util.isEqualOrAfter
+import no.nav.familie.ef.sak.util.isEqualOrBefore
 import org.springframework.stereotype.Service
 import java.math.BigDecimal
 import java.math.RoundingMode
+import java.time.LocalDate
 
 @Service
 class BeregningService {
 
-    fun beregnFullYtelse(beregningRequest: BeregningRequest): List<Beløpsperiode> {
-        return finnGrunnbeløpsPerioder(beregningRequest.stønadFom, beregningRequest.stønadTom).map {
-            Beløpsperiode(it.fraOgMedDato,
-                          it.tilDato,
-                          it.beløp.multiply(BigDecimal(2.25)).divide(BigDecimal(12)).setScale(0, RoundingMode.HALF_UP))
+    fun beregnYtelse(beregningRequest: BeregningRequest): List<Beløpsperiode> {
+        val vedtaksperioder = beregningRequest.vedtaksperiode
+        val vedtaksperiodeTilDato = beregningRequest.vedtaksperiode.maxOf { it.tildato }
+
+        val mapTilInntektMedPeriode: List<Inntektsperiode> = beregningRequest.inntektsperioder
+                .mapIndexed { index, inntektsperiode ->
+                    if (index < beregningRequest.inntektsperioder.lastIndex && beregningRequest.inntektsperioder.size > 1) {
+                        Inntektsperiode(inntekt = inntektsperiode.inntekt,
+                                        samordningsfradrag = inntektsperiode.samordningsfradrag,
+                                        startDato = inntektsperiode.startDato,
+                                        sluttDato = beregningRequest.inntektsperioder[index + 1].startDato.minusDays(
+                                               1))
+                    } else {
+                        Inntektsperiode(inntekt = inntektsperiode.inntekt,
+                                        samordningsfradrag = inntektsperiode.samordningsfradrag,
+                                        startDato = inntektsperiode.startDato,
+                                        sluttDato = vedtaksperiodeTilDato)
+
+                    }
+
+                }
+
+        validerInnteksperioder(mapTilInntektMedPeriode, vedtaksperioder)
+
+        validerVedtaksperioder(vedtaksperioder)
+
```

Comment:
Ska vi validere att inntekten er mindre enn 7.1 G (eller vad grensen er for overgansstønad)
Ska vi sjekke att datumen for vedtaksperiode > 1967 og < getÅr().+3?